### PR TITLE
fix(Scripts/EmeraldDragons): Lethon should use shadow bolt whirl

### PR DESCRIPTION
**Pull request by temperrr**
(Thanks to Nefertumm for some help :) )

## Changes Proposed:
-  Added aura script

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11890

## SOURCE:

Cmangos https://github.com/cmangos/mangos-classic/commit/26edc21e23df03629872f45fa362a1ccc9672ea0

## Tests Performed:

- Tested

## How to Test the Changes:

1. Go to Lethon with 2 or 3 chars
2. Engage
3. notice that they will shoot shadowbolts, but only 1 side at a time and rotating (So left right left right)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/shadowboltwhirl.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO spell_script_names (spell_id, ScriptName) VALUES
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/shadowboltwhirl.sql:1]:
```suggestion
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
```

File: data/sql/updates/pending_db_world/shadowboltwhirl.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+    (24834, 'aura_shadow_bolt_whirl');
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/shadowboltwhirl.sql:2]:
```suggestion
    (24834, 'aura_shadow_bolt_whirl');
    
```

File: src/server/scripts/World/boss_emerald_dragons.cpp [hunk 3]
```diff
@@ -709,6 +693,42 @@ class MarkOfNatureTargetSelector
...
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHADOW_BOLT_WHIRL });
+    }
+
+    void HandlePeriodic(AuraEffect const* aurEff)
+    {
+        
+        Unit* caster = GetCaster();
```
Comment (Nefertumm) [inline src/server/scripts/World/boss_emerald_dragons.cpp:708]:
```suggestion
    {
        Unit* caster = GetCaster(); 
```

File: data/sql/updates/pending_db_world/shadowboltwhirl.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+    (24834, 'aura_shadow_bolt_whirl');
+
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/shadowboltwhirl.sql:3]:
```suggestion

```

File: data/sql/updates/pending_db_world/shadowboltwhirl.sql [hunk 5]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(24834, 'aura_shadow_bolt_whirl');
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/shadowboltwhirl.sql:2]: should be spell_xxx regardless of being an aura

**Review by Nyeriah — APPROVED** [review]

File: src/server/scripts/World/boss_emerald_dragons.cpp [hunk 6]
```diff
@@ -709,6 +701,41 @@ class MarkOfNatureTargetSelector
...
+    }
+
+    void HandlePeriodic(AuraEffect const* aurEff)
+    {
+        Unit* caster = GetCaster();
+        Unit* target = GetTarget();
+
+        if (!caster || !target)
+            return;
+        std::array<uint32, 8> spellForTick = { 24820, 24821, 24822, 24823, 24835, 24836, 24837, 24838 };
```
[the remark this answers is not in the document]
Comment (temperrr) [reply src/server/scripts/World/boss_emerald_dragons.cpp:720]: Oh yep that makes sense

**Comment by Maelthyrr** [comment]
Tested and work fine. Good job :D

**Pull request merged**